src/frontend/specification/ParameterTable.jl

```diff
@@ -1,120 +1,149 @@
-        :identifier,
-    ]
-    existing_columns = [haskey(partable.columns, key) for key in relevant_columns]
+    relevant_columns =
+        [:from, :relation, :to, :free, :value_fixed, :start, :estimate, :se, :param]
+    shown_columns = filter!(
+        col -> haskey(partable.columns, col) && length(partable.columns[col]) > 0,
+        relevant_columns,
+    )
 
-    as_matrix =
-        hcat([partable.columns[key] for key in relevant_columns[existing_columns]]...)
+    as_matrix = mapreduce(col -> partable.columns[col], hcat, shown_columns)
     pretty_table(
         io,
         as_matrix,
-        header = (
-            relevant_columns[existing_columns],
-            eltype.([partable.columns[key] for key in relevant_columns[existing_columns]]),
-        ),
+        header = (shown_columns, [eltype(partable.columns[col]) for col in shown_columns]),
         tf = PrettyTables.tf_compact,
     )
 
-    if haskey(partable.variables, :latent_vars)
-        print(io, "Latent Variables:    $(partable.variables[:latent_vars]) \n")
-    end
-    if haskey(partable.variables, :observed_vars)
-        print(io, "Observed Variables:  $(partable.variables[:observed_vars]) \n")
-    end
+    print(io, "Latent Variables:    $(partable.latent_vars) \n")
+    print(io, "Observed Variables:  $(partable.observed_vars) \n")
 end
 
 ############################################################################################
 ### Additional Methods
 ############################################################################################
 
 # Iteration --------------------------------------------------------------------------------
+ParameterTableRow = @NamedTuple begin
+    from::Symbol
+    relation::Symbol
+    to::Symbol
+    free::Bool
+    value_fixed::Any
+    param::Symbol
+end
 
-Base.getindex(partable::ParameterTable, i::Int) = (
-    partable.columns[:from][i],
-    partable.columns[:parameter_type][i],
-    partable.columns[:to][i],
-    partable.columns[:free][i],
-    partable.columns[:value_fixed][i],
-    partable.columns[:identifier][i],
+Base.getindex(partable::ParameterTable, i::Integer) = (
+    from = partable.columns[:from][i],
+    relation = partable.columns[:relation][i],
+    to = partable.columns[:to][i],
+    free = partable.columns[:free][i],
+    value_fixed = partable.columns[:value_fixed][i],
+    param = partable.columns[:param][i],
 )
 
-function Base.length(partable::ParameterTable)
-    len = missing
-    for key in keys(partable.columns)
-        len = length(partable.columns[key])
-        break
-    end
-    return len
-end
+Base.length(partable::ParameterTable) = length(partable.columns[:param])
+Base.eachindex(partable::ParameterTable) = Base.OneTo(length(partable))
+
+Base.eltype(::Type{<:ParameterTable}) = ParameterTableRow
+Base.iterate(partable::ParameterTable) = iterate(partable, 1)
+Base.iterate(partable::ParameterTable, i::Integer) =
+    i > length(partable) ? nothing : (partable[i], i + 1)
```

Comment:
You can get away with this :)
```suggestion
Base.iterate(partable::ParameterTable, i::Integer = 1) =
    i > length(partable) ? nothing : (partable[i], i + 1)
```